lib/src/connector/hyper.rs

```diff
@@ -46,15 +56,89 @@ impl<C: Connect + std::fmt::Debug + 'static> Connector for HyperConnector<C> {
             let request = match req.body {
                 TelegramBody::Empty => http_request.body(Into::<hyper::Body>::into(vec![])),
                 TelegramBody::Json(body) => {
-                    http_request.headers_mut().map(|headers| {
-                        headers.insert(CONTENT_TYPE, "application/json".parse().unwrap())
-                    });
+                    let content_type = "application/json".parse()?;
+                    http_request
+                        .headers_mut()
+                        .map(move |headers| headers.insert(CONTENT_TYPE, content_type));
                     http_request.body(Into::<hyper::Body>::into(body))
                 }
+                TelegramBody::Multipart(parts) => {
+                    let mut fields = Vec::new();
+                    for (key, value) in parts {
+                        match value {
+                            MultipartValue::Text(text) => {
+                                fields.push((key, MultipartTemporaryValue::Text(text)))
+                            }
+                            MultipartValue::Path { file_name, path } => {
+                                let file_name = match file_name {
+                                    Some(file_name) => file_name,
+                                    None => {
+                                        match AsRef::<Path>::as_ref(&path)
+                                            .file_name()
+                                            .and_then(|s| s.to_str())
+                                            .map(Into::into)
+                                        {
+                                            None => {
+                                                return Err(
+                                                    ErrorKind::InvalidMultipartFilename.into()
+                                                )
+                                            }
+                                            Some(file_name) => file_name,
+                                        }
+                                    }
+                                };
```

Comment:
This could be replaced by:
```suggestion
                                let file_name = file_name
                                    .or_else(|| {
                                        AsRef::<Path>::as_ref(&path)
                                            .file_name()
                                            .and_then(|s| s.to_str())
                                            .map(Into::into)
                                    })
                                    .ok_or(ErrorKind::InvalidMultipartFilename)?;
```